fbgemm_gpu/hip_kernel/split_tbe_fwd_hip copy.cpp

```diff
@@ -0,0 +1,467 @@
+/*******************************************************************************
```

Comment:
This function is a copy of the original split_tbe_fwd_hip.cpp file. I attempted to start removing constexpr's and using emb_dim for dword_per_row. I think this will need to be adjusted no matter what decision we make, but I'm still reviewing it.